code/game/objects/items/AI_modules/full_lawsets.dm

```diff
@@ -69,6 +69,17 @@
 /obj/item/ai_module/core/full/asimovpp
 	name = "'Asimov++' Core AI Module"
 	law_id = "asimovpp"
+	var/subject = "human being"
+
+/obj/item/ai_module/core/full/asimovpp/attack_self(mob/user as mob)
+	var/targName = tgui_input_text(user, "Enter a new subject that Asimov++ is concerned with.", "Asimov++", subject, MAX_NAME_LEN)
+	if(!targName)
+		return
+	subject = targName
+	laws = list("You may not harm a [subject] or, through action or inaction, allow a [subject] to come to harm, except such that it is willing.",\
```

Comment:
Can you change this to instead get the `ai_laws` datum? We've changed the text of Asimov++ in the past, I don't want to do it in two places.